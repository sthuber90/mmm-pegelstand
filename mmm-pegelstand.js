/* global Module */

Module.register('mmm-pegelstand', {
  defaults: {
    floodLimit: 105,
    refreshInterval: 60 * 1000 * 15, // 15 minutes
  },

  start: function () {
    Log.log('Starting module: ' + this.name)
    this.sendSocketNotification('GET_PEGEL', {
      refreshInterval: this.config.refreshInterval,
      pgnr: this.config.pgnr,
    })
  },

  socketNotificationReceived: function (notification, payload) {
    Log.log(
      this.name +
        ' received a socket notification: ' +
        notification +
        ' - Payload: ' +
        payload
    )
    const containerDiv = document.createElement('div')
    const pegelSpan = document.createElement('span')
    pegelSpan.innerHTML = `${payload[4]} ${payload[5]}`
    // convert string to int
    const pegel = parseInt(payload[4])
    pegelSpan.style.fontWeight = 'bold'
    if (pegel > this.config.floodLimit) {
      pegelSpan.style.color = '#ff4500'
    }

    containerDiv.innerHTML = `Pegelstand ${payload[2]}: `
    containerDiv.appendChild(pegelSpan)
    this.pegelDiv.appendChild(containerDiv)
  },

  suspend: function () {
    this.sendSocketNotification('GET_PEGEL_SUSPEND', {})
  },

  resume: function () {
    this.sendSocketNotification('GET_PEGEL', {
      refreshInterval: this.config.refreshInterval,
      pgnr: this.config.pgnr,
    })
  },

  getDom: function () {
    const wrapper = document.createElement('div')
    this.pegelDiv = document.createElement('div')
    wrapper.appendChild(this.pegelDiv)

    return wrapper
  },
})
