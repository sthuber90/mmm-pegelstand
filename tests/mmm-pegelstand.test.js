describe('mmm-pegelstand', () => {
  beforeAll(() => {
    require('../__mocks__/Logger')
    require('../__mocks__/Module')
  })

  const name = 'mmm-pegelstand'

  let MMMPeglstand

  beforeEach(() => {
    jest.resetModules()
    require('../mmm-pegelstand')

    MMMPeglstand = global.Module.create(name)
    MMMPeglstand.setData({ name, identifier: `Module_1_${name}` })
  })

  it('should have correct defaults', () => {
    expect(MMMPeglstand.defaults).toMatchSnapshot()
  })

  it('should start', () => {
    MMMPeglstand.start()
    expect(global.Log.log).toHaveBeenNthCalledWith(
      1,
      'Starting module: mmm-pegelstand'
    )
  })

  it('should get styles', () => {
    expect(MMMPeglstand.getStyles()).toEqual([])
  })

  it('should get scripts', () => {
    expect(MMMPeglstand.getScripts()).toEqual([])
  })

  it('should get DOM', () => {
    expect(MMMPeglstand.getDom()).toMatchSnapshot()
  })
})
