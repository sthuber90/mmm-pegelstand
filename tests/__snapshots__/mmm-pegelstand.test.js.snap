// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mmm-pegelstand should get DOM 1`] = `
<div>
  <div />
</div>
`;

exports[`mmm-pegelstand should have correct defaults 1`] = `
Object {
  "floodLimit": 105,
  "refreshInterval": 900000,
}
`;
